src/Umbraco.Web.UI.Client/src/views/languages/edit.controller.js

```diff
@@ -65,18 +63,40 @@
 
                     vm.loading = false;
                     makeBreadcrumbs();
-
-                }, 1000);
+                });
             }
         }
 
         function save() {
-            vm.page.saveButtonState = "busy";
-            // fake saving the language
-            $timeout(function(){
-                vm.page.saveButtonState = "success";
-                notificationsService.success(localizationService.localize("speechBubbles_languageSaved"));
-            }, 1000);
+
+            if (formHelper.submitForm({ scope: $scope, statusMessage: "Saving..." })) {
+                vm.page.saveButtonState = "busy";
+
+                languageResource.save(vm.language).then(function (lang) {
+                    vm.language = lang;
+                    vm.page.saveButtonState = "success";
+                    notificationsService.success(localizationService.localize("speechBubbles_languageSaved"));
+
+                    back();
```

Comment:
Intentional UX change @Shazwazza - to go back to the full list once we save?
I have no problem with it, but it's different to what we know currently as users.

If so is the `vm.page.saveButtonState = "success"` necessary if we never get to see it because we switch views?

Reply:
@warrenbuckley It seemed more natural to do this, else we need to hide the drop down when the language is created and there doesn't seem like much else to do. If you think it's not good, then please fix up, i'll leave it up to your creative mind to decide :)

Reply:
Nah it's all good @Shazwazza  when testing it out - it worked fine & felt natural 👍